src/data/senoppfolging/useVurderSenOppfolgingKandidat.ts

```diff
@@ -0,0 +1,30 @@
+import { useValgtPersonident } from "@/hooks/useValgtBruker";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { ISMEROPPFOLGING_ROOT } from "@/apiConstants";
+import { post } from "@/api/axios";
+import {
+  SenOppfolgingKandidatResponseDTO,
+  SenOppfolgingVurderingRequestDTO,
+} from "@/data/senoppfolging/senOppfolgingTypes";
+import { senOppfolgingKandidatQueryKeys } from "@/data/senoppfolging/useSenOppfolgingKandidatQuery";
+
+export const useVurderSenOppfolgingKandidat = (kandidatUuid: string) => {
+  const personident = useValgtPersonident();
+  const queryClient = useQueryClient();
+  const path = `${ISMEROPPFOLGING_ROOT}/senoppfolging/kandidater/${kandidatUuid}/vurderinger`;
+  const postVurdering = (vurdering: SenOppfolgingVurderingRequestDTO) =>
+    post<SenOppfolgingKandidatResponseDTO>(path, vurdering, personident);
+
+  return useMutation({
+    mutationFn: postVurdering,
+    onSuccess: (data: SenOppfolgingKandidatResponseDTO) => {
+      queryClient.setQueryData(
+        senOppfolgingKandidatQueryKeys.senOppfolgingKandidat(personident),
+        (oldData: SenOppfolgingKandidatResponseDTO[]) => [
+          data,
+          ...oldData.slice(1),
```

Comment:
Og da vet vi at `oldData[0]` vil alltid være den som har blitt oppdatert?

Reply:
Ja, skal være den nyeste kandidaten som vurderes (den som ligger først i lista). Blir uansett gjort nytt GET-kall når man refresher siden som gir "sannheten", så dette er bare for å slippe unødvendig kall her siden vi får oppdatert kandidat i responsen fra POST-kallet.